app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -0,0 +1,46 @@
+module ServiceTemplateTransformationPlan::ValidateConfigInfo
+  extend ActiveSupport::Concern
+
+  def validate_config_info(options)
+    self.class.validate_config_info(options)
+  end
+
+  module ClassMethods
+    def validate_config_info(options)
+      config_info = options[:config_info]
+
+      mapping = if config_info[:transformation_mapping_id]
+                  TransformationMapping.find(config_info[:transformation_mapping_id])
+                else
+                  config_info[:transformation_mapping]
+                end
+
+      raise _('Must provide an existing transformation mapping') if mapping.blank?
+
+      pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
+      post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
```

Comment:
I believe that @michaelkro might be including other details in the `config_info` including the OSP security group and the OSP flavor.

Though it might need to be in the `actions` section below.  Be sure to follow up and see where that needs to be added.

Reply:
Currently we have an outstanding UI PR - https://github.com/ManageIQ/manageiq-v2v/pull/658 where we are adding `osp_security_group` and `osp_flavor` to `actions` as part of the POST body for `service_templates`

@bzwei Is there anything that needs to be done here with regard to the 2 new additions - `osp_security_group` and `osp_flavor` ?

Reply:
`osp_` options do not impact `actions`, so no need to update `actions`
They will need to be updated in `service_resource` per vm. I would recommend another PR for both add/update such options.